🐛 BUG: Error when trying to use wrangler init from dash

### Which Cloudflare product(s) does this pertain to?

Workers for Platforms, Workers Runtime, Other

### What version(s) of the tool(s) are you using?

3.6.0

### What version of Node are you using?

19

### What operating system are you using?

Mac

### Describe the Bug

```
wrangler init --from-dash moneytalks

 ⛅️ wrangler 3.6.0
------------------
Using npm as package manager.
▲ [WARNING] The `init --from-dash` command is no longer supported. Please use `npm create cloudflare@2 moneytalks -- --type pre-existing --existing-script moneytalks` instead.

  The `init` command will be removed in a future version.


Running `npm create cloudflare@2 moneytalks -- --type pre-existing --existing-script moneytalks`...

create-cloudflare [args]

Positionals:
  name                                                                  [string]

Options:
      --type                                                            [string]
      --framework                                                       [string]
      --deploy                                                         [boolean]
      --ts                                                             [boolean]
      --git                                                            [boolean]
      --open             opens your browser after your deployment, set --no-open
                          to disable                   [boolean] [default: true]
  -y, --accept-defaults                                                [boolean]
      --version          Show version number                           [boolean]
  -h, --help             Show help                                     [boolean]

Too many positional arguments provided
npm ERR! code 1
npm ERR! path /Users/vawogbemi/Documents/moneytalks
npm ERR! command failed
npm ERR! command sh -c create-cloudflare moneytalks pre-existing moneytalks

npm ERR! A complete log of this run can be found in: /Users/vawogbemi/.npm/_logs/2023-08-29T18_14_33_671Z-debug-0.log
Error
npm ERR! code 1
npm ERR! path /Users/vawogbemi/Documents/moneytalks
npm ERR! command failed
npm ERR! command sh -c create-cloudflare moneytalks --type pre-existing --existing-script moneytalks

npm ERR! A complete log of this run can be found in: /Users/vawogbemi/.npm/_logs/2023-08-29T18_14_30_686Z-debug-0.log

✘ [ERROR] Command failed with exit code 1: npm create cloudflare@2 moneytalks -- --type pre-existing --existing-script moneytalks


If you think this is a bug then please create an issue at https://github.com/cloudflare/workers-sdk/issues/new/choose

```


### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

_No response_

FWIW I also get this error, and when I try the suggested alternative:

```
npm create cloudflare@2 worker-name -- --type pre-existing --existing-script worker-name
```

I get this error: 

```
create-cloudflare [args]

Positionals:
  name                                                                  [string]

Options:
      --type                                                            [string]
      --framework                                                       [string]
      --deploy                                                         [boolean]
      --ts                                                             [boolean]
      --git                                                            [boolean]
      --open             opens your browser after your deployment, set --no-open
                          to disable                   [boolean] [default: true]
  -y, --accept-defaults                                                [boolean]
      --version          Show version number                           [boolean]
  -h, --help             Show help                                     [boolean]

Too many positional arguments provided
npm ERR! code 1
npm ERR! path /Users/zander/code/otter
npm ERR! command failed
npm ERR! command sh -c create-cloudflare worker-name pre-existing worker-name

npm ERR! A complete log of this run can be found in: /Users/zander/.npm/_logs/2023-10-05T16_44_07_604Z-debug-0.log
Error
npm ERR! code 1
npm ERR! path /Users/zander/code
npm ERR! command failed
npm ERR! command sh -c create-cloudflare masto-faves-to-otter --type pre-existing --existing-script masto-faves-to-otter

npm ERR! A complete log of this run can be found in: /Users/zander/.npm/_logs/2023-10-05T16_44_05_926Z-debug-0.log
```

It seems that there's no way to init a worker that has been created on the dashboard or already exists on the dashboard..

--- a/.changeset/yellow-cycles-fix.md
+++ b/.changeset/yellow-cycles-fix.md
@@ -1,0 +1,5 @@
+---
+"create-cloudflare": minor
+---
+
+Fixes an issue that was causing the auto-update check not to run

--- a/packages/create-cloudflare/src/cli.ts
+++ b/packages/create-cloudflare/src/cli.ts
@@ -51,7 +51,13 @@ const isUpdateAvailable = async () => {
 // Spawn a separate process running the most recent version of c3
 export const runLatest = async () => {
 	const args = process.argv.slice(2);
-	await runCommand(`${npm} create cloudflare@latest -- ${args.join(" ")}`);
+
+	// the parsing logic of `npm create` requires `--` to be supplied
+	// before any flags intended for the target command.
+	const argString =
+		npm === "npm" ? `-- ${args.join(" ")}` : `${args.join(" ")}`;
+
+	await runCommand(`${npm} create cloudflare@latest ${argString}`);
 };
 
 // Entrypoint to c3

--- a/packages/create-cloudflare/src/cli.ts
+++ b/packages/create-cloudflare/src/cli.ts
@@ -51,7 +51,7 @@ const isUpdateAvailable = async () => {
 // Spawn a separate process running the most recent version of c3
 export const runLatest = async () => {
 	const args = process.argv.slice(2);
-	await runCommand(`${npm} create cloudflare@latest ${args.join(" ")}`);
+	await runCommand(`${npm} create cloudflare@latest -- ${args.join(" ")}`);
 };
 
 // Entrypoint to c3

--- a/packages/create-cloudflare/src/helpers/args.ts
+++ b/packages/create-cloudflare/src/helpers/args.ts
@@ -39,6 +39,12 @@ export const parseArgs = async (argv: string[]): Promise<Partial<C3Args>> => {
 			alias: "y",
 			type: "boolean",
 		})
+		.option("auto-update", {
+			type: "boolean",
+			default: C3_DEFAULTS.autoUpdate,
+			description:
+				"Automatically uses the latest version of `create-cloudflare`. Set --no-auto-update to disable",
+		})
 		.option("wrangler-defaults", { type: "boolean", hidden: true })
 		.version(version)
 		// note: we use strictOptions since `strict()` seems not to handle `positional`s correctly

--- a/packages/create-cloudflare/src/workers.ts
+++ b/packages/create-cloudflare/src/workers.ts
@@ -23,6 +23,7 @@ import {
 	npmInstall,
 	runCommand,
 } from "helpers/command";
+import { detectPackageManager } from "helpers/packages";
 import {
 	chooseAccount,
 	gitCommit,
@@ -33,6 +34,8 @@ import {
 	setupProjectDirectory,
 } from "./common";
 import type { C3Args, PagesGeneratorContext as Context } from "types";
+
+const { dlx } = detectPackageManager();
 
 export const runWorkersGenerator = async (args: C3Args) => {
 	const { name, path } = setupProjectDirectory(args);
@@ -120,7 +123,7 @@ async function copyExistingWorkerFiles(ctx: Context) {
 			join(tmpdir(), "c3-wrangler-init--from-dash-")
 		);
 		await runCommand(
-			`npx wrangler@3 init --from-dash ${ctx.args.existingScript} -y --no-delegate-c3`,
+			`${dlx} wrangler@3 init --from-dash ${ctx.args.existingScript} -y --no-delegate-c3`,
 			{
 				silent: true,
 				cwd: tempdir, // use a tempdir because we don't want all the files

